akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/HandlerBuilders.scala

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.javadsl
+
+import java.util.Objects
+import java.util.function.{ BiFunction, Predicate, Supplier }
+
+import akka.annotation.InternalApi
+import akka.persistence.typed.internal.EffectImpl
+import akka.persistence.typed.javadsl.HandlerBuilders.CommandHandlerBuilderByState.CommandHandlerCase
+import akka.util.OptionVal
+
+object HandlerBuilders {
+
+  final class CommandHandlerBuilder[Command, Event, State] @InternalApi private[persistence] () {
+
+    private var builders: List[CommandHandlerBuilderByState[Command, Event, State, State]] = Nil
+
+    protected def forState(statePredicate: Predicate[State]): CommandHandlerBuilderByState[Command, Event, State, State] = {
+      val builder = CommandHandlerBuilderByState.builder[Command, Event, State](statePredicate)
+      builders = builder :: builders
+      builder
+    }
+
+    protected def forNullState(): CommandHandlerBuilderByState[Command, Event, State, State] = {
+      val builder = CommandHandlerBuilderByState.builder[Command, Event, State](s ⇒ Objects.isNull(s))
+      builders = builder :: builders
+      builder
+    }
+
+    protected def forType[S <: State](stateClass: Class[S]): CommandHandlerBuilderByState[Command, Event, S, State] = {
```

Comment:
I made if `forType` because it's very strange to write `forState(Foo.class)` :-)

There are three methods:
* forType
* forNullState (is null predicate)
* forState (arbritaty predicate)

Reply:
sounds good, possibly `forType` could be even more clear with `forStateType`